### PR TITLE
Save the server! Fix for Region validation.

### LINKED ISSUE #23: Region search: Validate input.
Right now whatever we input into the box is submitted to the sever. When the input is invalid we don't get verbose or meaningful error messages back, so we should validate the regions clientside before submission.

Demonstrative example: 
example region search input

The 'ss' appended to the end of the input in that screenshot results in a 500 from the server, and no results for any of the regions.

### PR DESCRIPTION
**Pull request by sehgalvibhor**
### Please describe your PR:
Closes #23 
Disables the button if the user does not follow the syntax for the input. This might be a sweet temporary fix, can be improvised in the near future based on subsequent discussion/review.

## Review checklist: 

 Before merging, confirm the following tasks have been executed

- [x] review a _minified_ build (e.g. `lein prod`, _not_ `lein figwheel`). 

Checked the following pages load results successfully and allow you to proceed to the results or report page:

- [ ] ID resolver + results preview
- [ ] Templates execute and show results
- [ ] Templates allow you to select lists AND type in identifiers
- [ ] Search (dropdown preview version)
- [ ] Search (full results page)
- [ ] Report page loads, including homologues and tools
- [ ] Region search
- [ ] Login and logout works for more than one user (use test user demo@intermine.org pw demo if needed)

This is not an exhaustive list - if you spot something else strange please bring it up!

### CODE REVIEW

File: src/cljs/bluegenes/pages/regions/views.cljs [hunk 1]
```diff
@@ -137,6 +137,17 @@
      [:div.feature-tree-container
       {:class (if @results "shrinkified")} [feature-types-tree]]]))
 
+(defn matches-regex?
+  [v regex]
```
Comment (yochannah) [inline src/cljs/bluegenes/pages/regions/views.cljs:141]: The regex works great! Could we add a docstring here explaining what the regex is checking for, in plain english? You can see an example of a docstring on line 152 below.

**Review by yochannah — CHANGES_REQUESTED** [review]
Hey @sehgalvibhor! Sorry for taking so long to test this one. The code you've implemented works great, but I think it needs a little bit more before we can merge it - my concern is that users who are trying to use the region search might not quite know _why_ the form is disabled, in its current state. It might be easier for them if we provide some feedback - perhaps a polite error message explaining the mandatory format and telling them which line is problematic, if there are multiple lines.

One option to implement this might be custom html5 form validation: https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation#Customized_error_messages but other reasonable implementation methods can be considered as well.

If you want to read more about feedback on errors: https://www.nngroup.com/articles/error-message-guidelines/

**Comment by sehgalvibhor** [comment]
Got it! I'll make the necessary changes and update the PR.

**Comment by yochannah** [comment]
Hey @sehgalvibhor! This PR has been around a while; I'm going to close it for now but if you wish to work on it further please feel free to re-open it.
